packages/paypal-commerce-integration/src/paypal-commerce/paypal-commerce-payment-strategy.ts

```diff
@@ -144,6 +187,64 @@ export default class PayPalCommercePaymentStrategy implements PaymentStrategy {
         return false;
     }
 
+    private prepareVaultedInstrumentPaymentPayload(
+        methodId: string,
+        paypalOrderId: string,
+        paymentData: VaultedInstrument & HostedInstrument,
+    ): Payment {
+        const { instrumentId, shouldSetAsDefaultInstrument } = paymentData;
+
+        return {
+            methodId,
+            paymentData: {
+                formattedPayload: {
+                    vault_payment_instrument: null,
+                    set_as_default_stored_instrument: shouldSetAsDefaultInstrument || false,
+                    device_info: null,
+                    method_id: methodId,
+                    paypal_account: {
+                        order_id: paypalOrderId,
+                    },
+                    bigpay_token: {
+                        token: instrumentId,
+                    },
+                },
+            },
+        };
+    }
+
+    private preparePaymentPayload(
+        methodId: string,
+        paypalOrderId: string,
+        paymentData: OrderPaymentRequestBody['paymentData'],
+    ): Payment {
+        const { getFieldsValues } = this.paypalcommerce || {};
+
+        const { shouldSaveInstrument = false, shouldSetAsDefaultInstrument = false } =
+            isHostedInstrumentLike(paymentData)
+                ? {
+                      ...paymentData,
+                      shouldSaveInstrument:
+                          getFieldsValues?.().shouldSaveInstrument ||
+                          paymentData.shouldSaveInstrument,
+                  }
+                : {};
```

Comment:
It might be better to update this part with 
```suggestion
        const { shouldSaveInstrument = false, shouldSetAsDefaultInstrument = false } = isHostedInstrumentLike(paymentData)
            ? paymentData
            : {};
```

Reply:
The same as in this answer: https://github.com/bigcommerce/checkout-sdk-js/pull/2352#discussion_r1498008514